### ISSUE TITLE
Setup on Mac M1 is broken with /usr/local/ path hardcoded

### ISSUE THREAD

**Issue by nmahadevuni** [facebookincubator/velox#11237]
### Problem description

I was running the setup-macos.sh script after rebasing to the latest. The `install_build_prerequisites` method has /usr/local/bin/ path hardcoded to check for ccache. This fails on Mac M1.

### System information

Velox System Info v0.0.2
Commit: 01f6fe7e236a890348e04209307c72d5f3386227
CMake Version: 3.30.5
System: Darwin-23.5.0
Arch: arm64
C++ Compiler: /Library/Developer/CommandLineTools/usr/bin/c++
C++ Compiler Version: 15.0.0.15000309
C Compiler: /Library/Developer/CommandLineTools/usr/bin/cc
C Compiler Version: 15.0.0.15000309
CMake Prefix Path: /Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr;/opt/homebrew;/usr/local;/usr;/;/opt/homebrew/Cellar/cmake/3.30.5;/usr/local;/usr/X11R6;/usr/pkg;/opt;/sw;/opt/local

### CMake log

```bash
+ mv ccache-4.10.2-darwin/ccache /usr/local/bin/
mv: rename ccache-4.10.2-darwin/ccache to /usr/local/bin/ccache: Permission denied
```

**Comment by majetideepak** [comment]
What is the permission sof your `/usr/local/bin` directory? Users should usually be able to move to that folder.

**Comment by assignUser** [comment]
@majetideepak is there a reason not to install ccache from brew on macos?

**Comment by majetideepak** [comment]
@assignUser ccache from brew brings in fmt11 which broke the build due to fmt11 headers being used along with the fmt10 that the scripts install. We removed ccache install from brew as a fix.
We later fixed the header include problem with #10920. We can revert to installing ccache from brew.
@czentgr Is it accurate that we can keep fmt11 now?

**Comment by czentgr** [comment]
@majetideepak Yes, with INSTALL_PREFIX now working properly we can add ccache back. I have it in my local env and have no problems with it and fmt11 being in homebrew.
